From: Mauro Carvalho Chehab <mchehab@redhat.com>
Date: Fri, 2 Jul 2010 23:13:48 -0400
Subject: [edac] add support for DDR3 at EDAC core
Message-id: <20100702201348.59d6acbb@pedra>
Patchwork-id: 26698
O-Subject: [RHEL5_BZ#468877 1/5] edac: add support for DDR3 at EDAC core
Bugzilla: 468877
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

Add registered and unregistered DDR3 memory at edac structs and code.

Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/edac/edac_mc.c b/drivers/edac/edac_mc.c
index eb9f73b..8f2efee 100644
--- a/drivers/edac/edac_mc.c
+++ b/drivers/edac/edac_mc.c
@@ -82,6 +82,8 @@ static const char *mem_types[] = {
 	[MEM_FB_DDR2] = "Fully-Buffered-DDR2",
 	[MEM_RDDR2] = "Registered-DDR2",
 	[MEM_XDR] = "XDR",
+	[MEM_DDR3] = "Unbuffered-DDR3",
+	[MEM_RDDR3] = "Registered-DDR3",
 };
 
 static const char *dev_types[] = {
diff --git a/drivers/edac/edac_mc.h b/drivers/edac/edac_mc.h
index 1c3f443..0622af1 100644
--- a/drivers/edac/edac_mc.h
+++ b/drivers/edac/edac_mc.h
@@ -147,6 +147,8 @@ enum mem_type {
 #define MEM_FLAG_FB_DDR2	BIT(MEM_FB_DDR2)
 #define MEM_FLAG_RDDR2		BIT(MEM_RDDR2)
 #define MEM_FLAG_XDR            BIT(MEM_XDR)
+#define MEM_FLAG_DDR3		BIT(MEM_DDR3)
+#define MEM_FLAG_RDDR3		BIT(MEM_RDDR3)
 
 /* chipset Error Detection and Correction capabilities and mode */
 enum edac_type {
